homeassistant/components/octoprint/camera.py

```diff
@@ -38,30 +38,39 @@ async def async_setup_entry(
         [
             OctoprintCamera(
                 camera_info,
-                coordinator.device_info,
+                coordinator,
                 device_id,
                 verify_ssl,
             )
         ]
     )
 
 
-class OctoprintCamera(MjpegCamera):
+class OctoprintCamera(CoordinatorEntity[OctoprintDataUpdateCoordinator], MjpegCamera):
     """Representation of an OctoPrint Camera Stream."""
 
     def __init__(
         self,
         camera_settings: WebcamSettings,
-        device_info: DeviceInfo,
+        coordinator: OctoprintDataUpdateCoordinator,
         device_id: str,
         verify_ssl: bool,
     ) -> None:
         """Initialize as a subclass of MjpegCamera."""
         super().__init__(
-            device_info=device_info,
+            coordinator=coordinator,
+        )
+        MjpegCamera.__init__(
+            self,
+            device_info=coordinator.device_info,
             mjpeg_url=camera_settings.stream_url,
             name="OctoPrint Camera",
             still_image_url=camera_settings.external_snapshot_url,
             unique_id=device_id,
             verify_ssl=verify_ssl,
         )
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return self.coordinator.last_update_success
```

Comment:
This is already the default provided by the CoordinatorEntity, and thus can be omitted.